### ISSUE TITLE
Highlighting applies to whole signcolumn?

### ISSUE THREAD

**Issue by skbolton** [lewis6991/gitsigns.nvim#54]
Are the highlights supposed to be applied to the whole signcolumn or just the text? It looks like in the gif of the readme just the `text` are colored. My situation has the whole sign column highlighted.

**To Reproduce**

init.vim:

```vim
let $PLUGIN_DIRECTORY = '~/.config/nvim/bundle'
set runtimepath^=$PLUGIN_DIRECTORY/plenary.nvim
set runtimepath^=$PLUGIN_DIRECTORY/gitsigns.nvim

lua << EOF
require('gitsigns').setup {
  signs = {
    add          = {hl = 'DiffAdd'   , text = '│', numhl='GitSignsAddNr'},
    change       = {hl = 'DiffChange', text = '│', numhl='GitSignsChangeNr'},
    delette       = {hl = 'DiffDelete', text = '_', numhl='GitSignsDeleteNr'},
    topdelete    = {hl = 'DiffDelete', text = '‾', numhl='GitSignsDeleteNr'},
    changedelete = {hl = 'DiffChange', text = '~', numhl='GitSignsChangeNr'},
  }
}
EOF
```
The above config is just copied from the default in the readme. I have tried defining it and and letting the defaults kick in. Same result.

![2021-02-09_23-03](https://user-images.githubusercontent.com/13260877/107472347-087bdb80-6b2c-11eb-9417-f7db2743e6f8.png)

**Additional context**
System: Linux
Neovim version: 0.5.0-dev+1059

**Comment by skbolton** [comment]
Immediately after posting this I figured out the issue. Putting solution here for anyone else who stumbles like I did. The `hl` option that you pass to the config is an entire highlight group and will grab its fg bg ect. Its not just the fg highlight. I was using `DiffAdd` as my highlight which is configured to do a green background and black foreground. By switching to a highlight like `GitGutterAdd` (lots of themes have git gutter support) I was able to get what I wanted since that highlight has the fg and background configured correctly.
